integrationTests/oneNodeNetwork.go

```diff
@@ -0,0 +1,99 @@
+package integrationTests
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/p2p"
+	"github.com/ElrondNetwork/elrond-go/process"
+)
+
+type oneNodeNetwork struct {
+	advertiser   p2p.Messenger
+	Nodes        []*TestProcessorNode
+	IdxProposers []int
+	Round        uint64
+	Nonce        uint64
+
+	Node *TestProcessorNode
+}
+
+// NewOneNodeNetwork creates a one-node network, useful for some integration tests
+func NewOneNodeNetwork() *oneNodeNetwork {
+	numOfShards := 1
+	nodesPerShard := 1
+	numMetachainNodes := 0
+
+	n := &oneNodeNetwork{
+		advertiser:   CreateMessengerWithKadDht(""),
+		IdxProposers: make([]int, numOfShards+1),
+	}
+
+	n.Nodes = CreateNodes(
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		GetConnectableAddress(n.advertiser),
+	)
+
+	for i := 0; i < numOfShards; i++ {
```

Comment:
OneNodeNetwork has exactly 1 shard and 1 node.

Reply:
done